src/app/page/card/card.component.html

```diff
@@ -1,5 +1,21 @@
 <div class="container" *ngIf="cards">
   <div class="container">
+    <div *ngIf="areCardsEnded === true" class="modal fade" id="modal" role="dialog" data-backdrop="static">
+      <div class="modal-dialog">
+        <div class="modal-content">
+          <div class="modal-header">
+            <h4 class="modal-title">The deck is over</h4>
+          </div>
+          <div class="modal-body">
+            <p>Great job. You`ve finished your task for today . Do you want to continue learning?</p>
+          </div>
+          <div class="modal-footer">
+            <button type="button" class="btn btn-default" data-dismiss="modal" routerLink="/main">Return to home page</button>
+            <button type="button" class="btn btn-default" data-dismiss="modal" (click)="getAdditionalCards()">Continue learning</button>
```

Comment:
It's wrong. You have to check here and inside ngOnInit methods if there are normal cards first, and only if there are not - get additional ones.